var/spack/repos/builtin/packages/cp2k/package.py

```diff
@@ -221,14 +241,18 @@ def edit(self, spec, prefix):
         if 'superlu-dist@4.3' in spec:
             ldflags.insert(0, '-Wl,--allow-multiple-definition')
 
-        # libint-1.x.y has to be linked statically to work around
-        # inconsistencies in its Fortran interface definition
-        # (short-int vs int) which otherwise causes segfaults at runtime
-        # due to wrong offsets into the shared library symbols.
-        libs.extend([
-            os.path.join(spec['libint'].libs.directories[0], 'libderiv.a'),
-            os.path.join(spec['libint'].libs.directories[0], 'libint.a'),
-        ])
+        if '@:6.9' in spec:
+            # libint-1.x.y has to be linked statically to work around
+            # inconsistencies in its Fortran interface definition
+            # (short-int vs int) which otherwise causes segfaults at runtime
+            # due to wrong offsets into the shared library symbols.
+            libs.extend([
+                os.path.join(spec['libint'].libs.directories[0], 'libderiv.a'),
+                os.path.join(spec['libint'].libs.directories[0], 'libint.a'),
+            ])
+        else:
+            fcflags += ['$(shell pkg-config --cflags libint2)']
+            libs += ['$(shell pkg-config --libs libint2)']
```

Comment:
That's hacky, but on the other hand `cp2k` recipe is. A better way to handle this would be if `libint` provides the flag itself - but we might improve this aspect in another PR.

Reply:
sure, we can add a `libs` property, although it should depend on the user of libint to determine whether to link statically or not (and for `libint-1.x` we had to link statically due to a broken interface)